### PR TITLE
IDEM-40

### PR DESCRIPTION
**Pull request by jrocca82**
Screenshot 2022-06-03 132424
Screenshot 2022-06-03 132438

### CODE REVIEW

**Comment by jrocca82** [comment]
Screenshot 2022-06-03 141214

**Review by JasminFarhi — DISMISSED** [review]

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 1]
```diff
@@ -82,6 +86,54 @@ const ClaimScreen: React.FC = () => {
     }
   };
 
+  const verifyEmail = async (email: string) => {
+    const armoredKey = await pgpLocalStorage.get();
+    try {
+      const uploadResponse = await axios.post<UploadPGPKeyResponse>(
+        "https://keys.openpgp.org/vks/v1/upload",
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/ClaimScreen.tsx:93]: Can we put url into config file same as ritestream?

File: app/src/navigation/screens/ClaimScreen.tsx [hunk 2]
```diff
@@ -82,6 +86,54 @@ const ClaimScreen: React.FC = () => {
     }
   };
 
+  const verifyEmail = async (email: string) => {
+    const armoredKey = await pgpLocalStorage.get();
+    try {
+      const uploadResponse = await axios.post<UploadPGPKeyResponse>(
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/ClaimScreen.tsx:92]: Can we use api same as ritestream instead?

File: app/src/navigation/screens/PGPScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      if (initialPgp) {
+        setKeytext(initialPgp.keytext);
+      }
+    })();
+  }, []);
+
+  const importPGP = async () => {
+    setKeytext(keytext);
+    await pgpLocalStorage.save({ keytext: keytext });
+    console.log(await pgpLocalStorage.get());
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/PGPScreen.tsx:22]: Remove?

File: app/src/navigation/screens/ClaimScreen.tsx (hunk 1 above)
Comment (bitcoinbrisbane) [reply app/src/navigation/screens/ClaimScreen.tsx:93]: Im happy for this to be hard coded for v 0.1

File: app/src/navigation/screens/PGPScreen.tsx (hunk 3 above)
Comment (bitcoinbrisbane) [reply app/src/navigation/screens/PGPScreen.tsx]: Fixed.

**Review by bitcoinbrisbane — APPROVED** [review]

File: app/src/components/UserDetailsHeader.tsx [hunk 4]
```diff
@@ -45,6 +50,7 @@ const UserDetailsHeader: React.FC = () => {
       <View style={styles.userDetails}>
         <DetailOrPlaceholder value={name} bold={true} placeholderWidth={90} />
         <DetailOrPlaceholder value={email} placeholderWidth={150} />
+        <Text onPress={() => navigation.navigate("PGP")}>Import a PGP</Text>
```
Comment (bitcoinbrisbane) [inline app/src/components/UserDetailsHeader.tsx:53]:
```suggestion
        <Text onPress={() => navigation.navigate("PGP")}>Import a PGP/GPG key pair</Text>
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
One line to change

**Comment by bitcoinbrisbane** [comment]
Few things

* Lets make the button full width like the other buttons "Upload document" etc.
* Change "Paste your PUBLIC key here" to "Paste your PUBLIC Key here" as we use Key not key in 2 other places
* Make the input box same width as the button
* Pin the input box to the top and make it double the height.

image

File: app/src/components/UserDetailsHeader.tsx (hunk 4 above)
Comment (bitcoinbrisbane) [reply app/src/components/UserDetailsHeader.tsx:53]: Fixed.

**Comment by bitcoinbrisbane** [comment]
Also, i can't work out how to go back or cancel out of this screen.

**Comment by jrocca82** [comment]
Screenshot 2022-06-06 094318

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**